canonical-web-and-design/react-components
# Migrate the contextual menu from maas-ui

## Done

- Migrate the button component to TypeScript.
- Migrate the contextual menu from maas-ui.
- Add auto window adjustment.
- Migrate to TypeScript.
- Update SearchAndFilter to use the modified contextual menu.

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Open storybook and check that the contextual menu displays.
- Adjust the window size and see that the menu adjust to fit (you'll need to make your window very narrow).
- Check that the external state SearchAndFilter demo still works (/?path=/story/searchandfilter--external-state).

## Fixes

Fixes: https://github.com/canonical-web-and-design/react-components/issues/215.
Fixes: https://github.com/canonical-web-and-design/react-components/issues/149.
